src/components/search/Search.js

```diff
@@ -1,16 +1,20 @@
-import React from "react";
+import React, {useState} from "react";
 import SearchInput from "./input/SearchInput";
 import Button from "../common/button/Button";
 import classes from "./Search.module.css";
+import {useNavigate, useParams, useSearchParams} from "react-router-dom";
 
 function Search() {
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+  const {searchQuery} = useParams();
+  const [search, setSearch] = useState(searchQuery ? searchQuery : "")
   return (
     <div className={classes.search}>
-      <SearchInput/>
-      <Button name="SEARCH" size="large" action={() => {
-      }}/>
+      <SearchInput search={search} setSearch={setSearch}/>
+      <Button name="SEARCH" size="large" action={(e) => navigate(`/search/${search}?${searchParams.toString()}`)}/>
```

Comment:
Не нужно использовать event, если ты к нему не обращаешься
```suggestion
      <Button name="SEARCH" size="large" action={() => navigate(`/search/${search}?${searchParams.toString()}`)}/>
```

Reply:
Done